appserver/payara-appserver-modules/yubikey-authentication/src/main/java/fish/payara/security/identitystores/YubikeyCredential.java

```diff
@@ -0,0 +1,81 @@
+ * 
+ *     The contents of this file are subject to the terms of either the GNU
+ *     General Public License Version 2 only ("GPL") or the Common Development
+ *     and Distribution License("CDDL") (collectively, the "License").  You
+ *     may not use this file except in compliance with the License.  You can
+ *     obtain a copy of the License at
+ *     https://github.com/payara/Payara/blob/master/LICENSE.txt
+ *     See the License for the specific
+ *     language governing permissions and limitations under the License.
+ * 
+ *     When distributing the software, include this License Header Notice in each
+ *     file and include the License file at glassfish/legal/LICENSE.txt.
+ * 
+ *     GPL Classpath Exception:
+ *     The Payara Foundation designates this particular file as subject to the "Classpath"
+ *     exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ *     file that accompanied this code.
+ * 
+ *     Modifications:
+ *     If applicable, add the following below the License Header, with the fields
+ *     enclosed by brackets [] replaced by your own identifying information:
+ *     "Portions Copyright [year] [name of copyright owner]"
+ * 
+ *     Contributor(s):
+ *     If you wish your version of this file to be governed by only the CDDL or
+ *     only the GPL Version 2, indicate your decision by adding "[Contributor]
+ *     elects to include this software in this distribution under the [CDDL or GPL
+ *     Version 2] license."  If you don't indicate a single choice of license, a
+ *     recipient has the option to distribute your version of this file under
+ *     either the CDDL, the GPL Version 2 or to extend the choice of license to
+ *     its licensees as provided above.  However, if you add GPL Version 2 code
+ *     and therefore, elected the GPL Version 2 license, then the option applies
+ *     only if the new code is made subject to such option by the copyright
+ *     holder.
+ *
+ */
+
+package fish.payara.security.identitystores;
+
+import javax.security.enterprise.credential.Credential;
+
+/**
+ * Class representing a Yubikey One Time Password (OTP)
+ * 
+ * @author Mark Wareham
+*/
+public class YubikeyCredential implements Credential {
+
+    public static final int PUBLIC_ID_LENGTH = 12;
+    private String oneTimePasswordString;
+
+    public YubikeyCredential(String oneTimePasswordString) {
+        this.oneTimePasswordString = oneTimePasswordString;
+    }
+    
+    public String getOneTimePasswordString() {
+        return oneTimePasswordString;
+    }
+
+    public void clearCredential (){
+        oneTimePasswordString=null;
+    }
+    
+    public String getPublicID(){
+        String fullKey = getOneTimePasswordString();
+        if(fullKey==null){
+            return null;
+        }
+        if (fullKey.isEmpty()){
+            return "";
+        }
+        if(fullKey.length()<12){
```

Comment:
fullKey.length<12 makes fulKey.isEmpty redundant as the same result would be returned by this one

Reply:
I was thinking of StringUtils.isEmpty() which checks whitespace insensitively